lib/res_extended.js

```diff
@@ -1,17 +1,194 @@
+                throw new Error("Items must be type array.")
+            }
+
+            res.data({items});
+
+            return res;
+        };
+
+        res.add_item = function (item) {
+            res.add_data('items', item);
+
+            return res;
+        };
+
+        res.meta = function (meta) {
+            if (typeof meta !== 'object') {
+                throw new Error("Meta must be type object.")
+            }
+            if (!meta.hasOwnProperty('code')) {
+                throw new Error("Meta must have property code.")
+            }
+
+            Object.assign(response, {meta});
+
+            return res;
+        };
+
+        res.add_error = function () {
+            let args = Array.prototype.slice.call(arguments);
+            let key = undefined;
+            let value = undefined;
+
+            if (args.length > 1) {
+                if (typeof args[0] !== 'string') {
+                    throw new Error('Error key should be type string.');
+                }
+
+                key = args[0];
+                value = args[1];
+            } else {
+                value = args[0];
+            }
+
+            if (typeof value !== 'object') {
+                throw new Error('Error must be type object.');
+            }
+
+            if (!value.hasOwnProperty('code')) {
+                throw new Error('Error must have property code.')
+            }
+
+            if (!value.hasOwnProperty('message')) {
+                throw new Error('Error must have property message.')
+            }
+
+            if (typeof response.errors === 'undefined') {
+                if (typeof key === 'undefined') {
+                    response.errors = [];
+                } else {
+                    response.errors = {};
+                }
+            }
+
+            if (typeof key === 'undefined') {
+                response.errors.push(value)
+            } else {
+                if (typeof response.errors[key] === 'undefined') {
+                    response.errors[key] = [];
+                }
+
+                response.errors[key].push(value);
+            }
+
+            return res;
+        };
+
+        res.limit = function (limit) {
+            if (typeof limit !== 'number') {
+                throw new Error("Limit must be type number.")
+            }
+
+            if (typeof response.data !== 'object') {
+                response.data = {};
+            }
+
+            Object.assign(response.data, {limit});
+
+            return res;
+        };
+
+        res.total = function (total) {
+            if (typeof total !== 'number') {
+                throw new Error("Total must be type number.")
+            }
+
+            if (typeof response.data !== 'object') {
+                response.data = {};
+            }
+
+            Object.assign(response.data, {total});
+
+            return res;
+        };
+
+
+        next();
+    };
 };
+
+function get_key_value() {
```

Comment:
awkward placement of function. you can just include it on the res object. res._get_key_value(); or put in util